### PR TITLE
Better daycycle with moon every 1st day in month

### PR DESCRIPTION
**Pull request by mpffh1**

### CODE REVIEW

File: src/environment/ssDaylight.lua [hunk 1]
```diff
@@ -160,24 +160,26 @@ function ssDaylight:generateAmbientCurve(nightEnd, dayStart, dayEnd, nightStart)
...
-    curve:addKeyframe({x = 0.020, y = 0.020, z = 0.032, time = (dayEnd + 4 * eveningStep) * 60}) -- ambient night
-    curve:addKeyframe({x = 0.024, y = 0.024, z = 0.036, time = (nightStart - 0.15 * eveningStep) * 60})
-    curve:addKeyframe({x = 0.072, y = 0.072, z = 0.088, time = (nightStart + 0.01) * 60}) -- light up ambient to compensate for switch to moonlight
-    curve:addKeyframe({x = 0.072, y = 0.072, z = 0.088, time = (nightStart + 0.02) * 60})
-    curve:addKeyframe({x = 0.020, y = 0.020, z = 0.032, time = (nightStart + 0.15 * eveningStep) * 60}) -- back to regular ambient settings
-    curve:addKeyframe({x = 0.020, y = 0.020, z = 0.032, time = 24.00 * 60})
-
+    curve:addKeyframe({x = 0.003, y = 0.003, z = 0.003, time = 0.00 * 60})
+    curve:addKeyframe({x = 0.003, y = 0.003, z = 0.003, time = 3.00 * 60})
+    curve:addKeyframe({x = 0.005, y = 0.005, z = 0.005, time = (nightEnd  - (2 * morningStep)) * 60}) --3:33 / 7:21
```
Comment (joskuijpers) [inline src/environment/ssDaylight.lua]: These times in the comment, for what day in the year are they?

File: src/environment/ssDaylight.lua [hunk 2]
```diff
@@ -187,51 +189,81 @@ function ssDaylight:generateSunColorCurve(nightEnd, dayStart, dayEnd, nightStart
...
-    curve:addKeyframe({x = 1.000, y = 0.900, z = 0.300, time = (dayEnd + 2 * eveningStep) * 60}) -- orange sunset
-    curve:addKeyframe({x = 0.900, y = 0.400, z = 0.100, time = (dayEnd + 3 * eveningStep) * 60}) -- shift to red
-    curve:addKeyframe({x = 0.300, y = 0.250, z = 0.500, time = (dayEnd + 4 * eveningStep) * 60}) -- shift to blue
-    curve:addKeyframe({x = 0.050, y = 0.060, z = 0.100, time = (nightStart - 0.15 * eveningStep) * 60})
-    curve:addKeyframe({x = 0.000, y = 0.000, z = 0.000, time = (nightStart + 0.01) * 60}) -- shift to brief complete darkness for switch to moonlight
-    curve:addKeyframe({x = 0.000, y = 0.000, z = 0.000, time = (nightStart + 0.02) * 60})
-    curve:addKeyframe({x = 0.050, y = 0.060, z = 0.100, time = (nightStart + 0.15 * eveningStep) * 60})
-    curve:addKeyframe({x = 0.060, y = 0.070, z = 0.150, time = 24.00 * 60}) -- shift to bluish darkness
+-------------------------	Moon
+local currentDay ,daysInSeason ,dayInYear ,seasonAtDay ,dayInSeason
```
Comment (joskuijpers) [inline src/environment/ssDaylight.lua]: Could you indent all these lines properly?
You are calculating if you should show the moon?

Comment (joskuijpers) [inline src/environment/ssDaylight.lua]: You could also make this all a function that determines if there is a moon :P `isNightWithMoon()`

Comment (mpffh1) [inline src/environment/ssDaylight.lua]: ok,
1.  i have indent  and commented some lines
2. I only ask the first day of the month
3. I'm not so good at programming see 2.
Took 2 days to find the right variable in yours code and calculate the daytimes (again and again with testing)
if you want to make a funktion to your code -- feel free to get my idea
ps.:please excuse my bad English

File: src/environment/ssDaylight.lua [hunk 3]
```diff
@@ -11,7 +11,7 @@ ssDaylight = {}
 g_seasons.daylight = ssDaylight
 
 function ssDaylight:load(savegame, key)
-    self.latitude = Utils.clamp(ssXMLUtil.getXMLFloat(savegame, key .. ".environment.latitude", 51.9), 0, 72)
+    self.latitude = Utils.clamp(ssXMLUtil.getXMLFloat(savegame, key .. ".environment.latitude", 47.4), 0, 72)
```
Comment (joskuijpers) [inline src/environment/ssDaylight.lua]: Why did you change this from the default we have? The latitude we had is from the same area we have growth and weather data from.

**Comment by joskuijpers** [comment]
If this is merged, #41 can be closed because it can't be reasonably done

File: src/environment/ssDaylight.lua (hunk 3 above)
Comment (mpffh1) [inline src/environment/ssDaylight.lua]: oh that´s my latitude
Was a test , whether the times adjust
and yes they do ,so everybody can put in his own latitude
i play a while and the weather data is good in game,that fits

**Comment by joskuijpers** [comment]
Alright, still haven't had time to carefully look at the code, but we decided to not include this in V1, because our deadline is near and we need to do a lot more testing.

Personally, I really like it :)

**Comment by joskuijpers** [comment]
This will need quite some rework, if at all possible, since the latest changes.
